l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -0,0 +1,1722 @@
+        province_model = self.env["res.country.state"]
+        return province_model.search(
+            [("code", "=", provinceCode), ("country_id.code", "=", "IT")]
+        )
+
+    def log_inconsistency(self, message):
+        inconsistencies = self.env.context.get("inconsistencies", "")
+        if inconsistencies:
+            inconsistencies += "\n"
+        inconsistencies += message
+        # convert to dict in order to be able to modify context
+        self.env.context = dict(self.env.context)
+        self.env.context.update(inconsistencies=inconsistencies)
+
+    def check_partner_base_data(self, partner_id, DatiAnagrafici):
+        partner = self.env["res.partner"].browse(partner_id)
+        if (
+            DatiAnagrafici.Anagrafica.Denominazione
+            and partner.name != DatiAnagrafici.Anagrafica.Denominazione
+        ):
+            self.log_inconsistency(
+                _("Company Name field contains '%s'." " Your System contains '%s'")
+                % (DatiAnagrafici.Anagrafica.Denominazione, partner.name)
+            )
+        if (
+            DatiAnagrafici.Anagrafica.Nome
+            and partner.firstname != DatiAnagrafici.Anagrafica.Nome
+        ):
+            self.log_inconsistency(
+                _("Name field contains '%s'." " Your System contains '%s'")
+                % (DatiAnagrafici.Anagrafica.Nome, partner.firstname)
+            )
+        if (
+            DatiAnagrafici.Anagrafica.Cognome
+            and partner.lastname != DatiAnagrafici.Anagrafica.Cognome
+        ):
+            self.log_inconsistency(
+                _("Surname field contains '%s'." " Your System contains '%s'")
+                % (DatiAnagrafici.Anagrafica.Cognome, partner.lastname)
+            )
+
+    def getPartnerBase(self, DatiAnagrafici):  # noqa: C901
+        if not DatiAnagrafici:
+            return False
+        partner_model = self.env["res.partner"]
+        cf = DatiAnagrafici.CodiceFiscale or False
+        vat = False
+        if DatiAnagrafici.IdFiscaleIVA:
+            # Format Italian VAT ID to always have 11 char
+            # to avoid validation error when creating the given partner
+            if DatiAnagrafici.IdFiscaleIVA.IdPaese.upper() == "IT":
+                vat = "{}{}".format(
+                    DatiAnagrafici.IdFiscaleIVA.IdPaese.upper(),
+                    DatiAnagrafici.IdFiscaleIVA.IdCodice.rjust(11, "0")[:11],
+                )
+            else:
+                vat = "{}{}".format(
+                    DatiAnagrafici.IdFiscaleIVA.IdPaese.upper(),
+                    re.sub(r"\W+", "", DatiAnagrafici.IdFiscaleIVA.IdCodice).upper(),
+                )
+        partners = partner_model
+        res_partner_rule = (
+            self.env["ir.model.data"]
+            .sudo()
+            .xmlid_to_object("base.res_partner_rule", raise_if_not_found=False)
+        )
+        if vat:
+            domain = [("sanitized_vat", "=", vat)]
```

Comment:
`sanitized_vat` alla v14 non esiste poiché lo aggiunge il modulo `base_vat_sanitized` in [OCA/partner-contact](https://github.com/OCA/partner-contact).
Inoltre questo modulo alla 14 è segnato come `obsoleto` ([Migration to version 14.0](https://github.com/OCA/partner-contact/issues/971))